tests/src/main/java/net/neoforged/neoforge/debug/transfer/VanillaHandlersTests.java

```diff
@@ -0,0 +1,566 @@
+                    helper.assertValueEqual(0, unequipEvents.getPlain(), "unequip event count");
+
+                    // Once we commit, the equip event should fire
+                    tx.commit();
+                    helper.assertValueEqual(1, equipEvents.getPlain(), "equip event count");
+                    helper.assertValueEqual(0, unequipEvents.getPlain(), "unequip event count");
+                }
+
+                try (var tx = Transaction.open(null)) {
+                    if (wrapper.extract(ItemResource.of(Items.DIAMOND_HORSE_ARMOR), 1, tx) != 1) {
+                        helper.fail("Should have extracted 1 diamond horse armor");
+                    }
+                    // No additional events yet - in case the extraction is rolled back
+                    helper.assertValueEqual(1, equipEvents.getPlain(), "equip event count");
+                    helper.assertValueEqual(0, unequipEvents.getPlain(), "unequip event count");
+
+                    // Once we commit, the unequip event should fire
+                    tx.commit();
+                    helper.assertValueEqual(1, equipEvents.getPlain(), "equip event count");
+                    helper.assertValueEqual(1, unequipEvents.getPlain(), "unequip event count");
+                }
+
+                helper.succeed();
+            });
+        });
+    }
+
+    @GameTest
+    @EmptyTemplate
+    @TestHolder(description = "Player armor wrapper only allows equippable items, and checks for curse of binding.")
+    public static void testPlayerArmorWrapper(ExtendedGameTestHelper helper) {
+        var creativePlayer = helper.makeMockPlayer();
+        var chestWrapper = PlayerInventoryWrapper.of(creativePlayer).getArmorSlot(EquipmentSlot.CHEST);
+
+        var diamondChestplate = ItemResource.of(Items.DIAMOND_CHESTPLATE);
+        var curseOfBinding = new ItemEnchantments.Mutable(ItemEnchantments.EMPTY);
+        curseOfBinding.set(helper.getHolder(Enchantments.BINDING_CURSE), 1);
+        var cursedDiamondChestplate = diamondChestplate.with(DataComponents.ENCHANTMENTS, curseOfBinding.toImmutable());
+
+        try (var tx = Transaction.open(null)) {
+            if (chestWrapper.insert(ItemResource.of(Items.DIAMOND_PICKAXE), 1, tx) != 0) {
+                helper.fail("Should have rejected diamond pickaxe as player armor");
+            }
+            if (chestWrapper.insert(diamondChestplate, 1, tx) != 1) {
+                helper.fail("Should have inserted 1 diamond chestplate");
+            }
+            if (chestWrapper.extract(diamondChestplate, 1, tx) != 1) {
+                helper.fail("Should have extracted 1 diamond chestplate");
+            }
+            if (chestWrapper.insert(cursedDiamondChestplate, 1, tx) != 1) {
+                helper.fail("Should have inserted 1 cursed diamond chestplate");
+            }
+            if (chestWrapper.extract(cursedDiamondChestplate, 1, tx) != 1) {
```

Comment:
Can we get a comment in the code to make it clearer that this is expected to work because it is a creative player? Or at least rename `chestWrapper` similar to the survival one to be `creativeChestWrapper`?